### DISCUSSION TITLE
求指导name 'load_model' is not defined这个报错怎么解决

### DISCUSSION THREAD

**Discussion by Zaneze2024** [BAAI-Agents/Cradle#75] [Q&A]
self.detect_model = load_model("./cache/GroundingDINO_SwinB_cfg.py", "./cache/groundingdino_swinb_cogcoor.pth")
NameError: name 'load_model' is not defined
这个报错怎么解决啊

**Comment by it666-png** [comment]
遇到相同的问题，如何解决呢

**Comment by turtoes** [comment]
https://github.com/IDEA-Research/GroundingDINO 先拉这个库

**Comment by WeihaoTan** [comment]
It seems that you did not download the .pth model file. Could you follow https://github.com/BAAI-Agents/Cradle/blob/main/docs/envs/rdr2.md to install grounding dino and see whether the error still exists?

**Reply by marco25cip** [reply]
hi, i get the same error, i download the .pth model file and putted in /cache directory, i follow the instructions for intstall groundingdino and still get the same error, could you help me please? thx 
![Screenshot 2024-07-30 152414](https://github.com/user-attachments/assets/1e695cf4-9601-407b-b6ef-a882325b6cc5)
![Screenshot 2024-07-30 152631](https://github.com/user-attachments/assets/43b940e6-6a27-403b-8e1e-5b1a2059c53a)

**Reply by marco25cip** [reply]
i 'fix' this error **replacing  def __init__(self)** in line 64 with **def load_model(self):**  in file located cradle/utilis/object_utilis.py, let me know if works for you guys!

**Comment by LeeOrange-is-me** [comment]
This error exsits in `cradle/utils/object_utils.py: line 7-10`
``` Python
try:
    from groundingdino.util.inference import load_model, load_image // run error.
except:
    pass
```
So we have to install `groundingdino`, Here is the solution:
``` Python
conda activate cradle-dev
git clone https://github.com/IDEA-Research/GroundingDINO.git
cd GroundingDINO/
```
then copy the GroundingDINO/groundingdino to the sys.path.
